app/controllers/users_controller.rb

```diff
@@ -1,9 +1,16 @@
+# frozen_string_literal: true
+
 class UsersController < ApplicationController
-	before_action :authenticate_user!, only: [:show]
+  before_action :authenticate_user!, only: %i[show index]
+
+
+  def index 
+  	@users = User.all
+  end
 
-	def show
+  def show
     @user = User.find(params[:id])
     @post = Post.new
-    @posts = Post.where("author_id" => @user.id)
+    @posts = Post.where('author_id' => @user.id)
```

Comment:
Please consider that, since the user has many posts, you are able to access the posts made by the user with something similar to `@posts = @user.posts`

Reply:
alright thanks for the tip